compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -293,7 +295,91 @@ def _do_logout_endpoints(
+
+    The force flag will overwrite an existing (non-directory) config.py.bak
+
+    This method uses a randomly generated intermediate output file in case
+    there are any permission or unforeseen file system issues
+    """
+    ep_dir = get_config_dir() / name
+    old_config = ep_dir / "config.py"
+    config_backup = ep_dir / "config.py.bak"
+
+    try:
+        # Scan config.py first in case it's a no-op
+        with open(old_config) as f:
+            lines = f.readlines()
+        reformatted_count = 0
+        output_lines = []
+        if lines:
+            for line in lines:
+                modified_line = False
+                for k, v in FUNCX_COMPUTE_IMPORT_UPDATES.items():
+                    if line.startswith(k):
+                        modified_line = True
+                        output_lines.append(line.replace(k, v, 1))
+                        break
+                if not modified_line:
+                    output_lines.append(line)
+                else:
+                    reformatted_count += 1
+        format_msg = f"No funcX import statements found in config.py for {name}"
+        if reformatted_count == 0:
+            return format_msg
+
+        remove_backup = False
+        if os.path.exists(config_backup):
+            if not force:
+                msg = (
+                    f"{config_backup} already exists.\n"
+                    "Rename it or use the --force flag to update config."
+                )
+                raise ClickException(msg)
+            if os.path.isdir(config_backup):
+                msg = (
+                    f"{config_backup} is a directory.\n"
+                    "Rename it before proceeding with config update."
+                )
+                raise ClickException(msg)
+            remove_backup = True
+
+        # Write to temporary file in case of unexpected file errors
+        tmp_output_path = ep_dir / ("config.py." + uuid.uuid4().hex)
+        with open(tmp_output_path.name, "w") as f:
+            for line in output_lines:
+                f.write(line)
+        if remove_backup:
+            os.remove(config_backup)
+        shutil.move(old_config, config_backup)  # Preserve file timestamp
+        os.rename(tmp_output_path.name, old_config)
+        format_msg = (
+            f"{reformatted_count} lines were modified for endpoint {name}\n"
+            f"  The previous config has been renamed to {config_backup}"
+        )
+        return format_msg
+
+    except FileNotFoundError as err:
+        msg = f"No config.py was found for endpoint ({name}) in {ep_dir}"
+        raise ClickException(msg) from err
+    except ClickException:
+        raise
+    except Exception as err:
+        msg = f"Unknown Exception {err} attempting to reformat config.py in {ep_dir}"
+        raise ClickException(msg) from err
+
+
+def read_config(endpoint_dir: pathlib.Path, warn_funcx_imports: bool = True) -> Config:
     endpoint_name = endpoint_dir.name
```

Comment:
I doesn't look like we actually use `warn_funcx_imports` at any call-sites.  Could this be removed?

Reply:
Originally put it in for potential tests that need it, but yes removing it til needed.